osf/models/preprint_provider.py

```diff
@@ -14,21 +14,19 @@
 
 class PreprintProvider(ObjectIDMixin, BaseModel):
```

Comment:
Please add two fields to the model:

- `allow_submissions`, boolean, default to true, all existing providers should be set to true;
- `additional_providers`, list of strings, will be used to allow the provider to mix in other SHARE-harvested preprint providers into their whitelist.

The API fields should be dev_only for the moment. This is to support https://openscience.atlassian.net/browse/EOSF-680?filter=18001